docs/source/conf.py

```diff
@@ -315,6 +315,16 @@
     ("py:class", "enum.Enum"),
     ("py:class", "bytes"),
     ("py:class", "bytearray"),
+    ("py:class", "numpy.dtype"),
+    ("py:class", "numpy.ndarray"),
+    ("py:class", "pandas.core.series.Series"),
+    ("py:class", "pandas.core.frame.DataFrame"),
+    ("py:class", "pyspark.sql.dataframe.DataFrame"),
+    ("py:class", "mlflow.types.schema.DataType"),
+    ("py:class", "mlflow.types.schema.ColSpec"),
+    ("py:class", "mlflow.types.schema.Schema"),
+    ("py:class", "mlflow.models.model.Model"),
+    ("py:class", "mlflow.models.signature.ModelSignature"),
```

Comment:
Adding these to  suppress warningss coming from type hints for model schema.

<img width="975" alt="Screen Shot 2020-12-03 at 23 00 57" src="https://user-images.githubusercontent.com/17039389/101043768-00e70600-35c2-11eb-92eb-641aebd705c1.png">

Reply:
The reason `mlflow.models.signature.ModelSignature` is treated as an invalid target is because we don't have `mlflow/models/signature.rst`.

Reply:
How often / in what situations will we need to add to this list?

Reply:
We'd need to add this list when we add type annotations.

```python
def new_api(x: dict):  # fine because dict is a python's built-in object
    ...

def new_api(x: sklearn.xxx.Yyy):  # emits a warning
    ...

def new_api(x: mlflow.xxx.yyy):  # might emit a warning if there isn't a page for mlflow.xxx.yyy.
    ...
```

Reply:
we could disable the warnings by setting `nitpicky` to `False`:

https://github.com/mlflow/mlflow/blob/master/docs/source/conf.py#L311